src/pages/index.js

```diff
@@ -212,44 +157,190 @@ function getCardElement(data) {
+  }
+
+  function setInitialLikeState(isLiked) {
+    if (isLiked) {
+      cardLikeButton.classList.add("card__content-like-button_active");
+    } else {
+      cardLikeButton.classList.remove("card__content-like-button_active");
+    }
+  }
+
+  // set initial like state from card data
+  const isLiked = data.isLiked || false;
+  setInitialLikeState(isLiked);
+
   return cardElement;
 }
 
-initialCards.forEach((element) => {
-  const cardElem = getCardElement(element);
-  cardsList.append(cardElem);
-});
+// ===== Handlers =====
+function submitEditProfileModal(evt) {
+  evt.preventDefault();
+  editProfileSaveBtn.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editProfileNameInput.value,
+      about: editProfileDescriptionInput.value,
+    })
+    .then((updatedUserInfo) => {
+      profileNameElement.textContent = updatedUserInfo.name;
+      profileDescriptionElement.textContent = updatedUserInfo.about;
+      closeModal(editProfileModal);
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      editProfileSaveBtn.textContent = "Save";
+    });
+}
 
-const modalPreview = document.querySelector("#preview-modal");
+function submitEditAvatarModal(evt) {
+  evt.preventDefault();
+  newAvatarSaveBtn.textContent = "Saving...";
+  api
+    .editUserAvatar({ avatar: editAvatarUrlInput.value })
+    .then((updatedAvatarInfo) => {
+      profileAvatarElement.src = updatedAvatarInfo.avatar;
+      evt.target.reset();
+      toggleButtonState([editAvatarUrlInput], evt.submitter, settings);
+      console.log("The current button state is {}");
+      closeModal(editAvatarModal);
+    })
+    .catch((err) => console.error("Error updating avatar:", err))
+    .finally(() => {
+      newAvatarSaveBtn.textContent = "Save";
+    });
+}
 
-const modalPreviewCloseBtn = modalPreview.querySelector(
-  ".modal__close-btn_preview"
-);
+function addCard(evt) {
+  evt.preventDefault();
+  modalSaveBtn.textContent = "Saving...";
+  api
+    .addNewCard({
+      name: newPostImageCaptionInput.value,
+      link: newPostImageLinkInput.value,
+    })
+    .then((data) => {
+      const newCard = getCardElement(data);
+      cardsList.prepend(newCard);
+      evt.target.reset();
+      toggleButtonState(
+        [newPostImageLinkInput, newPostImageCaptionInput],
+        evt.submitter,
+        settings
+      );
+      closeModal(newPostModal);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      modalSaveBtn.textContent = "Save";
+    });
+}
 
-const modalPreviewBackground = modalPreview.querySelector(".modal-background");
+function handleDeleteSubmit() {
+  deleteCardButton.textContent = "Deleting...";
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteCardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      deleteCardButton.textContent = "Delete";
+    });
+}
 
```

Comment:
* The `delete modal` is not visible on 320px (mobile) 
* Please, make the **avatar** match the design, regardless of the picture **size**  https://snipboard.io/jPLMHD.jpg . It should be always `round` and with a fixed width
* Please, make the `avatar` match the design on 320px (mobile) https://snipboard.io/r7HBOx.jpg . It should be in the center and round. And the `edit avatar` button should be visible on 320px